### PR TITLE
open zfs_clone_010_pos.ksh to verify zfs clones property displays right

### PR DESCRIPTION
**Pull request by liaoyuxiangqin**
information：

open zfs_clone_010_pos.ksh in zfs_clone Test group,and because the macro ZFS_MAXPROPLEN used
in function print_dataset which size is 4096, so modify the clone number and truncated length for clone list,
after modify this script test passed, thanks!

### CODE REVIEW

File: tests/runfiles/linux.run [hunk 1]
```diff
@@ -74,11 +74,10 @@ tests = ['zfs_001_neg', 'zfs_002_pos', 'zfs_003_neg']
 
 # DISABLED:
 # zfs_clone_005_pos - busy unmount
-# zfs_clone_010_pos - needs investigation
 [tests/functional/cli_root/zfs_clone]
 tests = ['zfs_clone_001_neg', 'zfs_clone_002_pos', 'zfs_clone_003_pos',
     'zfs_clone_004_pos', 'zfs_clone_006_pos',
-    'zfs_clone_007_pos', 'zfs_clone_008_neg', 'zfs_clone_009_neg']
+    'zfs_clone_007_pos', 'zfs_clone_008_neg', 'zfs_clone_009_neg', 'zfs_clone_010_pos']
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: style - 80 character line limit.

File: tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh [hunk 2]
```diff
@@ -219,14 +219,14 @@ i=1
...
-while((i < 7)); do
+while((i < 21)); do
 	log_must $ZFS clone $fs@snap $fs/$TESTCLONE$(python -c 'print "x" * 200').$i
 	((i=i+1))
 	((j=j+200))
 done
 clone_list=$($ZFS list -o clones $fs@snap)
 char_count=$($ECHO "$clone_list" | $TAIL -1 | wc | $AWK '{print $3}')
-[[ $char_count -eq 1024 ]] || \
+[[ $char_count -eq 4096 ]] || \
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh]: Just to be clear, this was caused by ZFS_MAXPROPLEN=1024 on illumos and ZFS_MAXPROPLEN=4096 on Linux.  Which is because MAXPATHLEN is defined differently.  I'd suggest reworking this slightly to use a variable to make this clear.  Something like:

```
if is_linux; then
        ZFS_MAXPROPLEN=4096        
else
        ZFS_MAXPROPLEN=1024
fi
```

Then use ZFS_MAXPROPLEN to calculate the number of iterations for the loop, say `$(( $ZFS_MAXPROPLEN/200+1 ))`.

**Comment by liaoyuxiangqin** [comment]
@behlendorf Thank you for your analysis of the script exist difference between zol with illumos  and giev fix guidance, at now the logic is more clear and flexible.

File: tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh [hunk 3]
```diff
@@ -217,16 +217,21 @@ log_note "verify clone list truncated correctly"
...
 fs=$TESTPOOL/$TESTFS1
+if is_linux; then
+	ZFS_MAXPROPLEN=4096
+else
+	ZFS_MAXPROPLEN=1024
+fi
 log_must $ZFS create $fs
 log_must $ZFS snapshot $fs@snap
-while((i < 7)); do
+while((i < $(( $ZFS_MAXPROPLEN/200+1 )))); do
```
Comment (rlaager) [inline tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh]: $(( $ZFS_MAXPROPLEN/200+1 )) evaluates to 6 when ZFS_MAXPROPLEN=1024. The old value was 7. Is this change intentional, and if so, why? (I don't know what this code does, so I'm not saying it is wrong. I am just asking as part of reviewing this.)

**Comment by rlaager** [comment]
The change to <= means the code runs the same number of times as before (on Illumos). I have no more objections.
